packages/blade/src/tokens/theme/__tests__/overrideTheme.test.ts

```diff
@@ -0,0 +1,131 @@
+import overrideTheme from '../overrideTheme';
+import paymentsTheme from '../paymentsTheme';
+import { Theme } from '../theme.d';
+
+const invalidOverridesObjectError = '[Blade:overrideTheme]: The overrides object is not valid';
+const invalidBaseThemeError =
+  '[Blade:overrideTheme]: The base theme provided is not a valid Blade theme';
+
+beforeAll(() => jest.spyOn(console, 'error').mockImplementation());
+afterAll(() => jest.restoreAllMocks());
+
+describe('overrideTheme', () => {
+  it('should return new theme based on overrides', () => {
+    const overrides = {
+      colors: {
+        brand: {
+          primary: {
+            300: {
+              onLight: 'someothercolor',
+            },
+          },
+        },
+        feedback: {
+          background: {
+            positive: {
+              highContrast: {
+                onLight: 'someothercolor',
+              },
+            },
+          },
+        },
+      },
+    };
+
+    const overridenTheme: Theme = {
```

Comment:
Subjective / optional: lot of nested spreading in this test object, can we create a deep copy of the default theme and then selectively modify only the `colors.brand.primary[300]` and `feedback.background.positive.highContrast` keys?

```ts
const overridenTheme = deepCopy(paymentsTheme);

overridenTheme.colors.brand.primary[300] = {
  ...paymentsTheme.colors.brand.primary[300],
  ...overrides.colors.brand.primary[300],
}
```

Does this convey the intent of test?